pkg/ee/default-policy-controller/controller.go

```diff
@@ -0,0 +1,399 @@
+		return fmt.Errorf("failed to list policy bindings: %w", err)
+	}
+
+	// Check if we already have a binding for this template
+	var currentPolicyBinding *kubermaticv1.PolicyBinding
+	for _, binding := range existingPolicyBindingList.Items {
+		if binding.Spec.PolicyTemplateRef.Name == template.Name {
+			// Check if this is an enforced or default binding
+			policyEnforced, _ := strconv.ParseBool(binding.Annotations[kubermaticv1.AnnotationPolicyEnforced])
+			policyDefault, _ := strconv.ParseBool(binding.Annotations[kubermaticv1.AnnotationPolicyDefault])
+
+			if policyEnforced || policyDefault {
+				currentPolicyBinding = &binding
+				break
+			}
+		}
+	}
+
+	reconcilers := []reconciling.NamedPolicyBindingReconcilerFactory{
+		PolicyBindingReconciler(r.log, template, currentPolicyBinding, cluster),
+	}
+	return reconciling.ReconcilePolicyBindings(ctx, reconcilers, namespace, r.Client)
+}
+
+func PolicyBindingReconciler(logger *zap.SugaredLogger, template kubermaticv1.PolicyTemplate, existingBinding *kubermaticv1.PolicyBinding, cluster *kubermaticv1.Cluster) reconciling.NamedPolicyBindingReconcilerFactory {
+	return func() (string, reconciling.PolicyBindingReconciler) {
+		bindingName := template.Name
+		return bindingName, func(binding *kubermaticv1.PolicyBinding) (*kubermaticv1.PolicyBinding, error) {
+			delete(template.Annotations, corev1.LastAppliedConfigAnnotation)
+
+			annotations := template.Annotations
+			if annotations == nil {
+				annotations = make(map[string]string)
+			}
+
+			if template.Spec.Enforced {
+				annotations[kubermaticv1.AnnotationPolicyEnforced] = strconv.FormatBool(true)
+			}
+			if template.Spec.Default {
+				annotations[kubermaticv1.AnnotationPolicyDefault] = strconv.FormatBool(true)
+			}
+
+			// Copy labels from the template if any
+			if len(template.Labels) > 0 {
+				if binding.Labels == nil {
+					binding.Labels = make(map[string]string)
+				}
+				for k, v := range template.Labels {
+					binding.Labels[k] = v
+				}
+			}
+
+			binding.Annotations = annotations
+			binding.Spec = kubermaticv1.PolicyBindingSpec{
+				PolicyTemplateRef: corev1.ObjectReference{
+					Name: template.Name,
+				},
+			}
+
+			logger.Infof("Reconciling PolicyBinding %s/%s", binding.Namespace, binding.Name)
+			return binding, nil
+		}
+	}
+}
+
+// isClusterTargeted checks if the PolicyTemplate targets the given cluster.
+func isClusterTargeted(cluster *kubermaticv1.Cluster, template *kubermaticv1.PolicyTemplate) bool {
+	// If no target is specified, we check the visibility
+	if template.Spec.Target == nil {
+		// Global policies apply to all clusters
+		if template.Spec.Visibility == kubermaticv1.PolicyTemplateVisibilityGlobal {
+			return true
+		}
+
+		// Project policies apply to clusters in the same project
+		if template.Spec.Visibility == kubermaticv1.PolicyTemplateVisibilityProject &&
+			template.Spec.ProjectID != "" &&
+			cluster.Labels[kubermaticv1.ProjectIDLabelKey] == template.Spec.ProjectID {
```

Comment:
Proposal: Put `projectID := cluster.Labels[kubermaticv1.ProjectIDLabelKey]` at the top of this function and it might become a bit more readable :)